storage/reads/modulo.go

```diff
@@ -0,0 +1,29 @@
+package reads
+
+func Modulo(dividend, modulus int64) int64 {
+	r := dividend % modulus
+	if r < 0 {
+		r += modulus
+	}
+	return r
+}
+
+func WindowStart(t, every, offset int64) int64 {
+	mod := Modulo(t, every)
+	off := Modulo(offset, every)
+	beg := t - mod + off
+	if mod < off {
```

Comment:
I stopped here for a moment to think about what is this case about, I guess a comment would be helpful.

Reply:
Added doc comments